main/go.mod

```diff
@@ -0,0 +1,7 @@
+module launch/main
+
+go 1.19
+
+replace packages/sorting => ../sorting
+
+require packages/sorting v0.0.0-00010101000000-000000000000
```

Comment:
I don't understand why you created sorting as a separate module. See https://github.com/golang-standards/project-layout#go-directories all what you need is `cmd` and either `internal` or `pkg`
You sorting is simple a package you import in the same module, no need to create a separate module for it. It's not only creates a mess, rather it introduces a dependency mismatch.
Good practice to have one module per repository. But again we don't need here 2 modules.